tools/wptrunner/wptrunner/wptrunner.py

```diff
@@ -158,9 +159,9 @@ def run_test_iteration(test_status, test_loader, test_source_kwargs, test_source
                        recording, test_environment, product, run_test_kwargs):
     """Runs the entire test suite.
     This is called for each repeat or retry run requested."""
-    tests = []
+    tests = defaultdict(list)
```

Comment:
Changed to tests_by_type, but kept it as a defaultdict because we used the 'extend' functions. Change to use assignment might be ok, but that will require some proof reading because that changed semantics.